From 86b2433b6f28983dcc340a820a7e47fe3aa9a432 Mon Sep 17 00:00:00 2001
From: Gibson Fahnestock <gib@uk.ibm.com>
Date: Fri, 6 Jan 2017 10:00:41 +0000
Subject: [PATCH] build: don't bundle deps/zlib if --shared-zlib set

Even if the --shared-zlib flag was used, the bundled deps/zlib was still
being compiled into the binary as it was required by the C++ test suite.

Fixes: https://github.com/nodejs/node/issues/10649
---
 node.gyp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/node.gyp b/node.gyp
index abba4c3..5075c51 100644
--- a/node.gyp
+++ b/node.gyp
@@ -900,7 +900,6 @@
             'HAVE_INSPECTOR=1',
           ],
           'dependencies': [
-            'deps/zlib/zlib.gyp:zlib',
             'v8_inspector_compress_protocol_json#host'
           ],
           'include_dirs': [
@@ -913,6 +912,11 @@
             'test/cctest/test_inspector_socket_server.cc'
           ],
           'conditions': [
+            [ 'node_shared_zlib=="false"', {
+              'dependencies': [
+                'deps/zlib/zlib.gyp:zlib',
+              ]
+            }],
             [ 'node_shared_openssl=="false"', {
               'dependencies': [
                 'deps/openssl/openssl.gyp:openssl'
